atlp-rwanda/script-heroes-bn-backend
# #174797538 CRUD operation on accommodations

#### What does this PR do?
Allow CRUD operation on accommodation by the admin
#### Description of Task to be completed?
- Create Tables for accommodations and rooms
- Create controllers for accommodations
- Create validation middleware for accommodations 
- Create routes for accommodations
#### How should this be manually tested?
- Clone this repo `git clone https://github.com/atlp-rwanda/script-heroes-bn-backend.git`
- Switch your working directory `cd script-heroes-bn-backend`
- Checkout this branch `git checkout ft-add-accommodation-details-#174797538`
- Install all dependencies `npm install`
- Migrate your database `npx sequelize db:migrate`
- Start the server `npm run dev`
- Hit on postman to test the following endpoints
    - `/api/accommodations`
        - *Post* and *Get* request
```json
{
    "facilityName": "facility Name",
    "locationId": 1,
    "description": "Full description of the facility",
    "roomType": "Type of room in facility",
    "photoUrl": "Url for image"
}
```
- `/api/accommodations/id`
    - Send a get request to get a single accommodation
    - Send a put request to update any field for the accommodation 
    - Send a delete request to delete an accommodation
- `/api/accommodations/id/rooms
    - Send a get request to get rooms in any accommodation with the provided id
- **Returned data for all accomodations should look like:**
```json
[
    {
        "id": 1,
        "facilityName": "Serena Hotel",
        "locationId": 1,
        "description": "This is Serena hotel",
        "createdAt": "2020-10-04T17:03:02.461Z",
        "updatedAt": "2020-10-05T06:13:29.628Z",
        "rooms": [
            {
                "id": 4,
                "UserId": null,
                "accomodationId": 1,
                "roomType": "Ordinary",
                "isVerified": false,
                "createdAt": "2020-10-05T06:02:20.366Z",
                "updatedAt": "2020-10-05T06:02:20.366Z"
            }
        ]
    },
    {
        "id": 5,
        "facilityName": "peace view Hotel",
        "location": "Gisenyi",
        "description": "This is Peace view hotel",
        "createdAt": "2020-10-05T06:26:49.215Z",
        "updatedAt": "2020-10-05T06:26:49.215Z",
        "rooms": [
            {
                "id": 5,
                "UserId": null,
                "accomodationId": 5,
                "roomType": null,
                "isVerified": false,
                "createdAt": "2020-10-05T06:26:49.325Z",
                "updatedAt": "2020-10-05T06:26:49.325Z"
            }
        ]
    }
]
```
#### Any background context you want to provide?
#### What are the relevant pivotal tracker stories?
[#174797538](https://www.pivotaltracker.com/story/show/174797538)
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/44426710/95049909-f6f07600-06ea-11eb-97d6-f384bb4e7196.png)
